app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -484,7 +487,7 @@ public String getDisplayRecipient(ZulipApp app) {
             ArrayList<String> names = new ArrayList<>();
 
             for (Person person : people) {
-                if (person.id != app.getYou().id || people.length == 1) {
+                if (!person.getEmail().equals(app.getYou().getEmail()) || people.length == 1) {
```

Comment:
I think the person objects all have id 0 because the user configuration response we receive is not deserialized into a person object with an id as it is supposed to be auto-generated.